### DISCUSSION TITLE
Fatture incloud sdk c#

### DISCUSSION THREAD

**Discussion by Fulvioavatar** [fattureincloud/api#291] [Q&A] [answered]
I'm not able to write code for obtaining company-id from List User Companies. Can you post an example in c#? I don't have find sites with examples, do you know someone?
I have obtained a manual token for authentication, and i wanna use it for perform List User Companies.
After obtaining company-id, someone may post some c# example of request to retrive invoices document data referred to a period?
Thanks.

**Comment by valmoz** [comment, marked as answer]
Hi @Fulvioavatar 
here I'll link some documentation that you could find useful.
You can also find some useful examples.

- [C# SDK Quickstart](https://developers.fattureincloud.it/docs/quickstarts/csharp-sdk-quickstart)
- [.NET SDK](https://developers.fattureincloud.it/docs/sdks/net-sdk)
- [Company ID](https://developers.fattureincloud.it/docs/basics/company-scoped-methods)
- [Manual Authentication](https://developers.fattureincloud.it/docs/authentication/manual-authentication)
- [Filters](https://developers.fattureincloud.it/docs/basics/additional-capabilities/filter-results)
- [Filter Helpers](https://developers.fattureincloud.it/docs/sdks/additional-capabilities/filter-results)
- [Vanilla Code Examples](https://developers.fattureincloud.it/docs/quickstarts/vanilla-code)
- [List User Companies](https://developers.fattureincloud.it/api-reference#get-/user/companies)
- [List Issued Documents](https://developers.fattureincloud.it/api-reference#get-/c/-company_id-/issued_documents)
- [Quickstarts GitHub Repo](https://github.com/fattureincloud/fattureincloud-quickstarts)
- [.NET SDK Readme](https://github.com/fattureincloud/fattureincloud-csharp-sdk/blob/master/README.md)

Please, notice that the examples you can find on these pages are generic, but you can adjust them to use the methods you need to recover the Company ID.

Thanks
Mauro

**Reply by Fulvioavatar** [reply]
Ok, after i solve mark as solution, thanks.

**Comment by Fulvioavatar** [comment]
I've just seen those pages, but for retriving company id i'm not able to write code, in the page https://developers.fattureincloud.it/api-reference/#get-/user/companies at the method List User Companies there is no code, only the result, so as i'm new to this sdk that i've istalled under visualstudio 2022, i'm not able to construct the right call, can you post directly the code to obtain company-id? Sorry for my dummy level!

**Reply by valmoz** [reply]
In the [C# Quickstart](https://developers.fattureincloud.it/docs/quickstarts/csharp-sdk-quickstart/) there's the List User Companies example...

**Reply by valmoz** [reply]
Also on the [SDK Documentation](https://github.com/fattureincloud/fattureincloud-csharp-sdk/blob/master/docs/UserApi.md#listusercompanies) there's an example

**Reply by Fulvioavatar** [reply]
Ok found and included in code with success, very thanks!!

**Reply by valmoz** [reply]
We also added SDK examples to the [Company ID guide](https://developers.fattureincloud.it/docs/basics/company-scoped-methods/).

**Reply by Fulvioavatar** [reply]
Very good thanks!
